File: server-Api/LifeIsGiving-Website2025/LifeIsGiving-Website2025/Models/Prize.cs
using LifeIsGiving_Website2025.Models.Enums;
using System.ComponentModel.DataAnnotations.Schema;

namespace LifeIsGiving_Website2025.Models
{
        [Table("Prizes")]
        public class Prize
        {
            public int Id { get; set; }
            public string Name { get; set; }     
            public string Description { get; set; }
            public PrizeCategory Category { get; set; }      
            public decimal Price { get; set; }   
            public string? ImageUrl { get; set; }   
            public int DonorId { get; set; }
            public User Donor { get; set; }
            public ICollection<Purchase> Purchases { get; set; } = new List<Purchase>();
        public bool CanPurchase { get; set; } = true;


    }

}

File: client-Angular/LifeIsGiving-Angular/src/app/core/models/Prize.ts
export interface Prize {
  id: number;
  name: string;
  description: string;
  category: number;
  price: number; // ← במקום ticketPrice
  imageUrl?: string;
  donorId: number;
  donorName?: string; // שדה נוסף להצגת שם התורם
  canPurchase: boolean;
}
